shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -783,36 +786,30 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
     DisposeViews();
 
     // Composite Platform Views.
-    for (auto view_id : views_to_recomposite) {
-      CompositeWithParams(view_id, current_composition_params[view_id]);
+    for (auto view_id : views_to_recomposite_) {
+      CompositeWithParams(view_id, current_composition_params_[view_id]);
     }
 
     for (const auto& cb : callbacks) {
       cb();
     }
 
-    // Create Missing Layers
-    for (auto i = 0u; i < missing_layer_count; i++) {
-      CreateLayer(gr_context,              //
-                  ios_context,             //
-                  MTLPixelFormatBGRA10_XR  //
-      );
-    }
-
     // Organize the layers by their z indexes.
-    auto active_composition_order = BringLayersIntoView(platform_view_layers, composition_order);
+    auto active_composition_order = BringLayersIntoView(platform_view_layers, composition_order_);
 
     // If a layer was allocated in the previous frame, but it's not used in the current frame,
     // then it can be removed from the scene.
-    RemoveUnusedLayers(unused_layers, composition_order, active_composition_order);
+    RemoveUnusedLayers(unused_layers, composition_order_, active_composition_order);
 
     // If the frame is submitted with embedded platform views,
     // there should be a |[CATransaction begin]| call in this frame prior to all the drawing.
     // If that case, we need to commit the transaction.
     [CATransaction commit];
-  });
+    latch.Signal();
+  };
 
   platform_task_runner_->PostTask(task);
+  latch.Wait();
```

Comment:
This seems necessary otherwise things quickly crash in release mode. I assume there is some state read on platform thread that is being touched in next preroll. On macOS we can get away with not blocking the raster thread because we have stricter separation of data. Possible could do it here too if the wait causes performance issues.